spec/unit/details.polyfill.spec.js

```diff
@@ -0,0 +1,78 @@
+/* global describe it expect beforeEach afterEach */
+
+var $ = window.jQuery
+
+describe('details-polyfill', function () {
+  'use strict'
+  var GOVUK = window.GOVUK
+
+  beforeEach(function (done) {
+    // Sample markup
+    this.$content = $(
+      '<details>' +
+      '<summary><span class="summary">Summary</span></summary>' +
+      '<div><p>Hidden content</p></div>' +
+      '</details>'
+    )
+
+    // Find elements
+    var $summaries = this.$content.find('summary')
+    var $hiddenContent = this.$content.find('div')
+
+    this.$summary1 = $summaries.eq(0)
+    this.$hiddenContent1 = $hiddenContent.eq(0)
+
+    // Add to page
+    $(document.body).append(this.$content)
+
+    setTimeout(function () {
```

Comment:
Essentially I needed to do asynchronous testing because of how the events are being chained.
I can avoid doing if like this, but I have to do more rewriting on the polyfill first. Let me know if you want me to try and get rid of the async test.

Reply:
👍